pkg/datapath/linux/config/config.go

```diff
@@ -400,7 +400,7 @@ func (h *HeaderfileWriter) WriteNodeConfig(w io.Writer, cfg *datapath.LocalNodeC
 		ctmap.WriteBPFMacros(fw, nil)
 	}
 
-	if option.Config.PolicyAuditMode {
+	if val, ok := option.Config.Opts.Opts[option.PolicyAuditMode]; ok && val != option.OptionDisabled {
```

Comment:
I don't understand this change as that's the only case we are accessing `Config.Opts.Opts` in `config.go`

Reply:
Hi, I added a comment to explain this change.

Reply:
Thanks! :pray:

Reply:
On a second look I think this specific change (using `Config.Opts.Opts[option.PolicyAuditMode]` in place of `option.Config.PolicyAuditMode`) should be moved into a separate commit as it's kind of unrelated to the more general change proposed here of regenerating the datapath program on agent config changes